src/evict/evict_lru.c

```diff
@@ -1521,7 +1521,18 @@ __wt_cache_eviction_worker(WT_SESSION_IMPL *session, int busy, int pct_full)
 				return (0);
 		}
 
-		/* Wait for the queue to re-populate before trying again. */
+		/*
+		 * Wait 1/10th second for the queue to re-populate before trying
+		 * again; if we've waited for a second, decrement the counter.
+		 * The decrement is defensive (otherwise the loop might never
+		 * terminate), and there have been bugs in other parts of the
+		 * code causing the wrong thread to infinitely loop here.
+		 */
+		if (++wait_count > 10) {
+			wait_count = 0;
+			if (--count == 0)
+				return (0);
+		}
```

Comment:
Well, the cache would only grow if there are no pages to evict in the queue, that is, the queue is empty for a second.

There was no workload that prompted this change, it's purely defensive in case the eviction server ever gets down here again with an empty queue.

I'm only interested in avoiding the infinite loop.

We could clear the wait-count if we find anything at all to evict, or we could increase the wait time to something much longer, maybe 10 seconds?